api/user.go

```diff
@@ -24,8 +24,8 @@ type ListUsersRequest struct {
 }
 
 type CreateUserRequest struct {
-	Name               string `json:"name" validate:"email,required"`
-	SetOneTimePassword bool   `json:"setOneTimePassword"`
+	Name       string `json:"name" validate:"email,required"`
+	ProviderID uid.ID `json:"providerID"` // optional. if missing, assume infra. This is needed for custom IDP implementations
```

Comment:
Nice to see `SetOneTimePassword` go.!

I'm confused as to why `ProviderID` is exposed here? Which custom IDP implementations require ProviderID? AFAIK no IDPs call this endpoint directly.

Reply:
all IDPs will have to know their own providerID from install/setup. It's not possible to look it up if you have multiple IDPs of the same type.
What Bruce said is correct; while we currently have no IDPs that call this, it's impossible to integrate with us without allowing an IDP to call this endpoint, so we should at least check that you intend to create a local Infra user vs something else.

Reply:
Let's remove this `ProviderID` in this PR until we need it? Calling this endpoint should be for creating a local user. Adding SCIM will be a whole other change

Reply:
SCIM aside, this is still important if you want to hook us up to your company's IDP. It's an optional field, so api consumers can ignore it, but the api is useless unless we already support your choice of IDP. That's not great API design.

Reply:
> How would a developer use infra login to login using that provider? Wouldn't infra login assume it was an oidc provider, attempt a login that way, and fail?

Right, good question. Today I suppose they'd still also have to be oidc compatible. If they're OIDC-compatible then JIT user creation will work fine. 

This isn't a user facing thing at the moment, it's really a question of getting the backend architecture right. Regardless of when you expose it, the endpoint and the API are specific to the Infra provider, and the code should reflect that, rather than assume.

Reply:
Could we remove the ambiguity with a comment? `CreateUserRequest`, and the `CreateUser` endpoint could both say "for the infra provider".